packages/logic-utils/src/utils/documentation.ts

```diff
@@ -0,0 +1,191 @@
+  const tagMatcherToReplacement: {
+    matcher: RegExp;
+    replacement: Record<ConvertTarget, string | ReplaceFunction>;
+  }[] = [
+    // Italics
+    {
+      matcher: /<i>(.+?)<\/i>/g,
+      replacement: { markdown: "*$1*", plaintext: "$1" },
+    },
+
+    // Bold
+    {
+      matcher: /<b>(.+?)<\/b>/g,
+      replacement: { markdown: "**$1**", plaintext: "$1" },
+    },
+    {
+      matcher: /<strong>(.+?)<\/strong>/g,
+      replacement: { markdown: "**$1**", plaintext: "$1" },
+    },
+
+    // Emphasis
+    {
+      matcher: /<em>(.+?)<\/em>/g,
+      replacement: { markdown: "***$1***", plaintext: "$1" },
+    },
+
+    // Headers
+    {
+      matcher: /<h1>(.+?)<\/h1>/g,
+      replacement: { markdown: "\n# $1\n\n", plaintext: "\n$1\n" },
+    },
+    {
+      matcher: /<h2>(.+?)<\/h2>/g,
+      replacement: { markdown: "\n## $1\n\n", plaintext: "\n$1\n" },
+    },
+    {
+      matcher: /<h3>(.+?)<\/h3>/g,
+      replacement: { markdown: "\n### $1\n\n", plaintext: "\n$1\n" },
+    },
+    {
+      matcher: /<h4>(.+?)<\/h4>/g,
+      replacement: { markdown: "\n#### $1\n\n", plaintext: "\n$1\n" },
+    },
+
+    // Lists
+    {
+      matcher: /<li>(.+?)<\/li>/g,
+      replacement: { markdown: "\n* $1", plaintext: "\n* $1" },
+    },
+    { matcher: /<ul>/g, replacement: { markdown: "", plaintext: "" } },
+    { matcher: /<\/ul>/g, replacement: { markdown: "\n\n", plaintext: "\n" } },
+    // TODO should we handle ordered lists (ol)?
+
+    // Code value
+    {
+      matcher: /<code>(.+?)<\/code>/g,
+      replacement: { markdown: "`$1`", plaintext: "$1" },
+    },
+
+    // Code block
+    {
+      matcher: /<pre>([^]+?)<\/pre>/g,
+      replacement: {
+        markdown: "\n```javascript\n$1```\n",
+        plaintext: "\n$1\n",
+      },
+    },
+
+    // Line break
+    { matcher: /<br\/>/g, replacement: { markdown: "\n", plaintext: "\n" } },
+    { matcher: /<br>/g, replacement: { markdown: "\n", plaintext: "\n" } },
+    { matcher: /<\/br>/g, replacement: { markdown: "\n", plaintext: "\n" } },
+
+    // HTML Escaping, e.g. "&lt;View" --> "<View"
+    // Note: this doesn't replace all html-encoded characters, only the most common ones
+    {
+      matcher: /.*/gm,
+      replacement: { markdown: unescape, plaintext: unescape },
+    },
+
+    // Links
+    // Assuming links are of the form: {@link <type>[ <text>]}
+    // Where the type doesn't contain whitespace, and neither the type nor text contain the "}" character
+    {
+      matcher: /{@link (([^\s}]+)\s)?([^}]+)}/g,
+      replacement: {
+        markdown: (all, _, type, text): string => {
+          return `[${text}](${getLink(model, type ?? text)})`;
+        },
+        plaintext: "$3",
+      },
+    },
+  ];
+
+  let contents = jsdoc;
+  forEach(tagMatcherToReplacement, (_) => {
+    const replacement = _.replacement[target];
+    // Workaround for typescript signature definition
+    if (typeof replacement === "string") {
```

Comment:
174 and 176 lines seem to do that same thing.
If there is a problem with string.prototype.definitions maybe we should solve it 
at design time with a cast instead of adding a runtime branch?

Reply:
The problem is that `replace` has 2 overloads: `(RegExp, string)` and `(RegExp, ReplaceFunction)`, instead of having the second argument as `string | ReplaceFunction`. Casting a `string | ReplaceFunction` to either one of them seems really wrong...

Reply:
o.k. but it is still just a types signature problem (design time) so we should not resolve it by changing the runtime.

I suggest using a cast / `// @ts-ignore` comment
- In TypeScript 3.9 we can use `// @ts-expect-error` comment
So when that signature is fixed we will get a compliation error to remove the comment.

- https://www.typescriptlang.org/docs/handbook/release-notes/overview.html#-ts-expect-error-comments

Reply:
Can also make a PR to fix it here:
- https://github.com/microsoft/TypeScript/blob/master/src/lib/es5.d.ts#L422-L434
- For OSS contribution goals :)

Reply:
According to ECMAScript specs it should be string | replaceFunc
- http://www.ecma-international.org/ecma-262/#sec-string.prototype.replace

Reply:
The contribution looks like a good idea and there is a good reason for this in the best practice guide (which is coincidentally exactly our use case):
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-union-types
I hope they will accept it.

Reply:
Cool, I will update to TypeScript 3.9 and add a `ts-expect-error` (different PR) so once
TypeScript@next is out with your fix, we will get a reminder to fix it in our code.

Reply:
> o.k. but it is still just a types signature problem (design time) so we should not resolve it by changing the runtime.
> 
> I suggest using a cast / `// @ts-ignore` comment
> 
> * In TypeScript 3.9 we can use `// @ts-expect-error` comment
>   So when that signature is fixed we will get a compilation error to remove the comment.
> * https://www.typescriptlang.org/docs/handbook/release-notes/overview.html#-ts-expect-error-comments

The problem with that is that if the type mismatch would not be because of the current reason (overload defined instead of a union type), and the types actually would not match, we won't get an error.
I'll change type type definition instead, I think it should work.